core/src/main/java/io/substrait/relation/ProtoRelConverter.java

```diff
@@ -527,6 +532,35 @@ private Rel newHashJoin(HashJoinRel rel) {
     return builder.build();
   }
 
+  private NestedLoopJoin newNestedLoopJoin(NestedLoopJoinRel rel) {
+    Rel left = from(rel.getLeft());
+    Rel right = from(rel.getRight());
+    Type.Struct leftStruct = left.getRecordType();
+    Type.Struct rightStruct = right.getRecordType();
+    Type.Struct unionedStruct = Type.Struct.builder().from(leftStruct).from(rightStruct).build();
+    var converter = new ProtoExpressionConverter(lookup, extensions, unionedStruct, this);
+    var builder =
+        NestedLoopJoin.builder()
+            .left(left)
+            .right(right)
+            .condition(
+                // defaults to true if the join expression is unassigned, resulting in a cartesian
+                // join
+                Optional.of(
+                    rel.hasExpression()
+                        ? converter.from(rel.getExpression())
+                        : Expression.BoolLiteral.builder().value(true).build()))
```

Comment:
One thing to note for NLJ is that the spec says the Join Expression is optional, but defaults to True which becomes a cartesian join. I've added the code to default to True here.